### PR TITLE
feat(apig/plugin): add a new resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support a new resource for APIG plugin policy.
For plugin policy type, we have support these kinds:
- [`CORS`](https://support.huaweicloud.com/intl/en-us/usermanual-apig/apig_03_0021.html)
- [`HTTP Response Header Management`](https://support.huaweicloud.com/intl/en-us/usermanual-apig/apig_03_0022.html)
- [`Request Throttling 2.0`](https://support.huaweicloud.com/intl/en-us/usermanual-apig/apig_03_0054.html)
- [`Kafka Log Push`](https://support.huaweicloud.com/intl/en-us/usermanual-apig/apig_03_0061.html)
- [`Circuit Breaker`](https://support.huaweicloud.com/intl/en-us/usermanual-apig/apig_03_0023.html)

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource support.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccPlugin_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccPlugin_basic -timeout 360m -parallel 4
=== RUN   TestAccPlugin_basic
=== PAUSE TestAccPlugin_basic
=== CONT  TestAccPlugin_basic
--- PASS: TestAccPlugin_basic (469.53s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig 469.604s
```
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccPlugin_httpResponse'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccPlugin_httpResponse -timeout 360m -parallel 4
=== RUN   TestAccPlugin_httpResponse
=== PAUSE TestAccPlugin_httpResponse
=== CONT  TestAccPlugin_httpResponse
--- PASS: TestAccPlugin_httpResponse (466.71s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig 466.785s
```
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccPlugin_rateLimit'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccPlugin_rateLimit -timeout 360m -parallel 4
=== RUN   TestAccPlugin_rateLimit
=== PAUSE TestAccPlugin_rateLimit
=== CONT  TestAccPlugin_rateLimit
--- PASS: TestAccPlugin_rateLimit (481.74s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      481.818s
```
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccPlugin_kafkaLog'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccPlugin_kafkaLog -timeout 360m -parallel 4
=== RUN   TestAccPlugin_kafkaLog
=== PAUSE TestAccPlugin_kafkaLog
=== CONT  TestAccPlugin_kafkaLog
--- PASS: TestAccPlugin_kafkaLog (678.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      678.350s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
The test result of `breaker` plugin is missing.
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccPlugin_breaker'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccPlugin_breaker -timeout 360m -parallel 4
=== RUN   TestAccPlugin_breaker
=== PAUSE TestAccPlugin_breaker
=== CONT  TestAccPlugin_breaker
--- PASS: TestAccPlugin_breaker (471.78s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      471.870s
```

File: docs/resources/apig_plugin.md [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+variable "connect_port" {}
+
+resource "huaweicloud_apig_plugin" "test" {
+  instance_id = var.instance_id
+  name        = var.plugin_name
+  type        = "kafka_log"
+  content     = jsonencode(
+    {
+      "broker_list": [for v in var.connect_addresses: format("%s:%d", v, var.connect_port)],
+      "topic": "${huaweicloud_dms_kafka_topic.test.name}",
```
Comment (deer-hang) [inline docs/resources/apig_plugin.md]:
```suggestion
      "topic": "${var.topic_name}",
```
add a variable "topic_name"

File: docs/resources/apig_plugin.md [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+
+* `id` - The ID of the plugin.
+
+* `created_at` - The creation time of the plugin.
+
+* `updated_at` - The latest update time of the plugin.
+
+## Import
+
+Plugins can be imported using their `id` and related dedicated instance ID, separated by a slash, e.g.
```
Comment (deer-hang) [inline docs/resources/apig_plugin.md]:
```suggestion
Plugins can be imported using their related dedicated instance ID and `id`, separated by a slash, e.g.
```

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_plugin_test.go [hunk 3]
```diff
@@ -0,0 +1,991 @@
...
+		},
+	})
+}
+
+func testAccPluginImportStateFunc(n string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return "", fmt.Errorf("resource (%s) not found: %s", n, rs)
+		}
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_plugin_test.go:87]: make variable values:
```
instanceID := rs.Primary.Attributes["instance_id"]
id := rs.Primary.Attributes["instance_id"]
```

File: docs/resources/apig_plugin.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/apig_plugin.md]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccPlugin_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccPlugin_ -timeout 360m -parallel 4
=== RUN   TestAccPlugin_basic
=== PAUSE TestAccPlugin_basic
=== RUN   TestAccPlugin_httpResponse
=== PAUSE TestAccPlugin_httpResponse
=== RUN   TestAccPlugin_rateLimit
=== PAUSE TestAccPlugin_rateLimit
=== RUN   TestAccPlugin_kafkaLog
=== PAUSE TestAccPlugin_kafkaLog
=== RUN   TestAccPlugin_breaker
=== PAUSE TestAccPlugin_breaker
=== CONT  TestAccPlugin_basic
=== CONT  TestAccPlugin_kafkaLog
=== CONT  TestAccPlugin_rateLimit
=== CONT  TestAccPlugin_breaker
--- PASS: TestAccPlugin_basic (466.70s)
=== CONT  TestAccPlugin_httpResponse
--- PASS: TestAccPlugin_breaker (469.78s)
--- PASS: TestAccPlugin_rateLimit (476.02s)
--- PASS: TestAccPlugin_kafkaLog (610.98s)
--- PASS: TestAccPlugin_httpResponse (462.95s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      929.714s
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_plugin.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	if err != nil {
+		return diag.Errorf("error deleting the plugin (%s): %s", pluginId, err)
+	}
+	return nil
+}
+
+func resourcePluginImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData,
+	error) {
+	importedId := d.Id()
+	parts := strings.SplitN(importedId, "/", 2)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apig/resource_huaweicloud_apig_plugin.go]:
```suggestion
	parts := strings.Split(importedId, "/", 2)
```

Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_plugin.go]: fixed.

File: docs/resources/apig_plugin.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/apig_plugin.md]: done.

**Pull request merged**